query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala

```diff
@@ -74,6 +94,39 @@ object RateFunctions {
     val result = if (isRate) (scaledDelta / (windowEnd - windowStart) * 1000) else scaledDelta
     result
   }
+
+  /**
+   * Calculates rate/delta/increase (period-counter) based on window information and between sample1 and sample2
+   *
+   * @param numSamples the number of samples inclusive of start and end
+   */
+  //scalastyle:off parameter.number
+  def extrapolatedPeriodicRate(windowStart: Long,
```

Comment:
We should not add extrapolation when calculating rate for period counters and histograms.
Adding extrapolation will result in some values being counted twice.

Reply:
Same with increase

Reply:
changed the implementation to reuse existing sumovertime and not extrapolate rate